lib/Dialect/Arc/ArcTypes.cpp

```diff
@@ -21,11 +21,17 @@ using namespace mlir;
 #define GET_TYPEDEF_CLASSES
 #include "circt/Dialect/Arc/ArcTypes.cpp.inc"
 
-unsigned StateType::getBitWidth() { return hw::getBitWidth(getType()); }
+unsigned StateType::getBitWidth() {
+  if (llvm::isa<seq::ClockType>(getType()))
+    return 1;
+  return hw::getBitWidth(getType());
+}
 
 LogicalResult
 StateType::verify(llvm::function_ref<InFlightDiagnostic()> emitError,
                   Type innerType) {
+  if (llvm::isa<seq::ClockType>(innerType))
+    return success();
```

Comment:
Not in `hw::getBitWidth` though :cry:

Reply:
Ah right, you're calling `hw::getBitWidth`. Why not `StateType::getBitWidth` though?

Reply:
Because that is not available at this point :cry: (it takes the type from the `this` parameter instead of a function argument). I should just factor this out into a static helper.

Reply:
Let me fix that post-merge.